lib/sycamore/sycamore/connectors/doc_reconstruct.py

```diff
@@ -0,0 +1,33 @@
+from abc import ABC, abstractmethod
+from typing import Callable
+
+from sycamore.data import Document
+
+
+class DocumentReconstructor(ABC):
+    @abstractmethod
+    def get_required_source_fields(self) -> list[str]:
+        pass
+
+    @abstractmethod
+    def get_doc_id(self, data) -> str:
```

Comment:
For now, I'd get rid of the abstract base class. It's not clear to me this is the right abstraction, but we won't know until we implement this for a few more databases.  E.g. it's assuming that there are source fields, and that the document comes back as a dictionary.

Reply:
Sure